### PR TITLE
amp-consent allow full screen after promptTrigger: action

### LINKED ISSUE #27819: amp-consent allow full screen after promptTrigger: action
[Link to original issue](https://github.com/ampproject/amphtml/issues/26197#issuecomment-598639396)

We had originally disallowed amp-consent to enlarge the 3p consent iframe, until user-interaction with the `promptUi` had been detected. 

It seems reasonable to also allow the 3p iframe to enlarge if the `promptTrigger: action`  signal is sent from a button to amp-consent as well.

/cc @zhouyx

### PR DESCRIPTION
**Pull request by micajuine-ho**
Keep track of `isActionPromptTrigger` when showing consentUI. 

`isActionPromptTrigger` was only used as part of clientInfo to be sent to iframe. 

Now it's also used to relax the restriction on `enter-fullscreen` signal sent from iframe. 

Closes #27819

### CODE REVIEW

File: extensions/amp-consent/0.1/test/test-consent-ui.js [hunk 1]
```diff
@@ -526,6 +527,79 @@ describes.realWin(
             'info': 'Entering fullscreen.',
           });
         });
+
+        describe('actionPromptTrigger', () => {
+          it('should not expand when actionPromptTrigger is false', async () => {
```
Comment (zhouyx) [inline extensions/amp-consent/0.1/test/test-consent-ui.js:532]: By default actionPromptTrigger would be false, this looks duplicate.

File: extensions/amp-consent/0.1/test/test-consent-ui.js [hunk 2]
```diff
@@ -526,6 +527,79 @@ describes.realWin(
...
+            expect(consentUI.isFullscreen_).to.be.true;
+            expect(consentUI.isActionPromptTrigger_).to.be.true;
+            expect(
+              consentUI.parent_.classList.contains(
+                consentUiClasses.iframeFullscreen
+              )
+            ).to.be.true;
+          });
+
+          it('should expand after hidden', async () => {
```
Comment (zhouyx) [inline extensions/amp-consent/0.1/test/test-consent-ui.js:575]: What's this test for? And can we remove it?

File: extensions/amp-consent/0.1/consent-ui.js [hunk 3]
```diff
@@ -225,7 +228,9 @@ export class ConsentUI {
     // Add to fixed layer
     this.baseInstance_.getViewport().addToFixedLayer(this.parent_);
     if (this.isCreatedIframe_) {
-      this.loadIframe_(isActionPromptTrigger).then(() => {
+      // Set this for every iframe request to show, no need to reset.
+      this.isActionPromptTrigger_ = isActionPromptTrigger;
```
Comment (zhouyx) [inline extensions/amp-consent/0.1/consent-ui.js:232]: I don't recommend caching the value because isActionPromptTrigger is tied to the `show()` action, not to the consentUI class. It's possible that `#show()` is called twice with different `isActionPromptTrigger` value. Caching it to the class, could lead to the value been overridden before the promise is resolved.
This won't lead to a race condition only because the notificationUIManager makes sure no two UI can be shown at the same time.

File: extensions/amp-consent/0.1/amp-consent.js [hunk 4]
```diff
@@ -239,6 +239,7 @@ export class AmpConsent extends AMP.BaseElement {
     });
 
     this.registerAction('prompt', () => {
+      console.log('register action prompt');
```
Comment (zhouyx) [inline extensions/amp-consent/0.1/amp-consent.js:242]: please remove

File: extensions/amp-consent/0.1/consent-ui.js (hunk 3 above)
Comment (zhouyx) [reply extensions/amp-consent/0.1/consent-ui.js:232]: Well I understand that you don't want to pass the variable to nested functions. I'm fine either way. But please make it clear that such race condition won't happen.

Comment (micajuine-ho) [reply extensions/amp-consent/0.1/consent-ui.js:232]: Ok, thanks for detailing your concern. I will add comments to explain that the race condition is not possible.

File: extensions/amp-consent/0.1/test/test-consent-ui.js (hunk 1 above)
Comment (micajuine-ho) [reply extensions/amp-consent/0.1/test/test-consent-ui.js:532]: fair point. Removing test.

File: extensions/amp-consent/0.1/test/test-consent-ui.js (hunk 2 above)
Comment (micajuine-ho) [reply extensions/amp-consent/0.1/test/test-consent-ui.js:575]: This is more like an e2e test to test a common workflow (consent prompt is shown on-load, user makes decision, user then clicks button that calls `consentUi#show(true)` consent prompt enters full screen). 

Do you think it should be removed?

Comment (zhouyx) [reply extensions/amp-consent/0.1/test/test-consent-ui.js:553]: I would rather have it removed. We are mocking many services in unit tests, a workflow doesn't make too much sense unless there's a corner case scenario or race condition you want to fake. 
My understanding is that everything before the second `consentUI.show(true)` doesn't affect the test at all. which makes it a duplicate test case.

Comment (micajuine-ho) [reply extensions/amp-consent/0.1/test/test-consent-ui.js:553]: SGTM
